03/tideman/tideman.c

```diff
@@ -0,0 +1,232 @@
+    for(int candidate_a = 0; candidate_a < candidate_count; candidate_a++) {
+
+        for(int candidate_b = candidate_a + 1; candidate_b < candidate_count; candidate_b++) {
+            preferences[ranks[candidate_a]][ranks[candidate_b]]++;
+        }
+    }
+
+    return;
+}
+
+// Record pairs of candidates where one is preferred over the other
+void add_pairs(void)
+{
+    for(int candidate_a = 0; candidate_a < candidate_count; candidate_a++) {
+
+        for(int candidate_b = candidate_a + 1; candidate_b < candidate_count; candidate_b++) {
+
+            if(preferences[candidate_a][candidate_b] > preferences[candidate_b][candidate_a]) {
+
+                pairs[pair_count].winner = candidate_a;
+                pairs[pair_count].loser = candidate_b;
+                pair_count++;
+            }
+            else if(preferences[candidate_a][candidate_b] < preferences[candidate_b][candidate_a]) {
+
+                pairs[pair_count].winner = candidate_b;
+                pairs[pair_count].loser = candidate_a;
+                pair_count++;
+            }
+        }
+    }
+
+    return;
+}
+
+// Sort pairs in decreasing order by strength of victory
+void sort_pairs(void)
+{
+    for(int outerCounter = 0; outerCounter < pair_count; outerCounter++) {
+
+        for(int innerCounter = outerCounter; innerCounter < pair_count; innerCounter++) {
+
+            int strengthCurrentPair = preferences[pairs[innerCounter].winner][pairs[innerCounter].loser];
+            int strengthNextPair = preferences[pairs[innerCounter + 1].winner][pairs[innerCounter + 1].loser];
+
+            if(strengthCurrentPair < strengthNextPair) {
+
+                pair temporaryPair; temporaryPair.winner = pairs[innerCounter].winner; temporaryPair.loser = pairs[innerCounter].loser;
+
+                pairs[innerCounter].winner = pairs[innerCounter + 1].winner;
+                pairs[innerCounter].loser = pairs[innerCounter + 1].loser;
+
+                pairs[innerCounter + 1].winner = temporaryPair.winner;
+                pairs[innerCounter + 1].loser = temporaryPair.loser;
+            }
+        }
+    }
+
+    return;
+}
+
+// Lock pairs into the candidate graph in order, without creating cycles
+void lock_pairs(void)
+{
+    for(int pairIndex = 0; pairIndex < pair_count; pairIndex++) {
+
+        if(isCycle(pairs[pairIndex].winner, pairs[pairIndex].loser) == false) {
+            locked[pairs[pairIndex].winner][pairs[pairIndex].loser] = true;
+        }
+    }
+
+    return;
+}
+
+// Print the winner of the election
+void print_winner(void)
+{
+    for(int counter = 0; counter < candidate_count; counter++) {
+        
+        bool winner = true;
+
+        for(int innerCounter = 0; innerCounter < candidate_count; innerCounter++) {
+
+            if(locked[innerCounter][counter] == true){
+                winner = false;
+                break;
+            }
+        }
+        if(winner == true) {
+            printf("Winner: %s\n", candidates[counter]);
+        }
```

Comment:
```suggestion
    for(int counter = 0; counter < candidate_count; counter++) {
        bool winner = true;

        for(int innerCounter = 0; innerCounter < candidate_count; innerCounter++) {
            if(locked[innerCounter][counter]){
                winner = false;
                break;
            }
        }
        
        if(winner) printf("Winner: %s\n", candidates[counter]);
```
Bei boolean Variablen reicht es wenn du die Variable als Bedingung schreibst. == true oder == false ist nicht nötig